Sources/SuperDelegate+StateRestoration.swift

```diff
@@ -26,22 +26,19 @@ import Foundation
 
 public protocol StateRestorationCapable: ApplicationLaunched {
     /// Called when your app can save its application state for restoration. Returns whether the app should save the current state.
-    @warn_unused_result
     func shouldSaveApplicationState(coder: NSCoder) -> Bool
     
     /// Called when your app can restore its application state after its interface has been loaded. Returns whether the app should restore.
-    @warn_unused_result
     func shouldRestoreApplicationState(coder: NSCoder) -> Bool
     
     /// Tells your delegate to save any high-level state information at the beginning of the state preservation process.
-    func willEncodeRestorableStateWithCoder(coder: NSCoder)
+    func willEncodeRestorableState(coder: NSCoder)
```

Comment:
hmm not sure if we wanna keep an argument label for coder... at least as `coder`.  Coder is impled with encode/decode.

NSObjects's newer APIs uses the verb instead of a noun..

Examples:

https://developer.apple.com/reference/objectivec/nsobject/1417074-awakeafter
https://developer.apple.com/reference/foundation/nscoding/1413933-encode